src/main/java/fi/aalto/cs/apluscourses/ui/base/TextField.java

```diff
@@ -0,0 +1,27 @@
+package fi.aalto.cs.apluscourses.ui.base;
+
+import fi.aalto.cs.apluscourses.ui.utils.TwoWayBindable;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import javax.swing.JTextField;
+
+/**
+ * A class deriving from {@link JTextField} that works with the {@link TwoWayBindable} class to
+ * reflect its text input to the source property.
+ */
+public class TextField extends JTextField {
+  public final transient TwoWayBindable<TextField, String> textBindable =
+      new TwoWayBindable<>(this, TextField::setText, TextField::getText);
+
+  /**
+   * Construct a TextField.
+   */
+  public TextField() {
+    addKeyListener(new KeyAdapter() {
```

Comment:
I guess this is fine. 👍

Reply:
Argh, I meant to use a document listener, this was a left-over. I'll change it

Reply:
`updateSource` is still additionally called when OK is selected in the dialog, so it should work, but document listener is clearly the more correct choice here